pkg/spacebinding/spacebindinglister_test.go

```diff
@@ -235,4 +235,199 @@ func TestNewSpaceBindingLister(t *testing.T) {
+			expectedErr           string
+		}{
+			"for Space-A": {
+				space:                 spaceA,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// we expect to have one spacebinding for the root space
+					*spacebinding.NewSpaceBinding(spaceAMur, spaceA, spaceA.Name, spacebinding.WithRole("admin")),
+				},
+			},
+			"for Space-B": {
+				space:                 spaceB,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// only expect space binding form space-B
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+				},
+			},
+			"for Space-C": {
+				space:                 spaceC,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// we expect spacebinding inherited from space-B
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+					// and specific one from space-C
+					*spacebinding.NewSpaceBinding(spaceCMur, spaceC, spaceC.Name, spacebinding.WithRole("maintainer")),
+				},
+			},
+			"for Space-D": {
+				space:                 spaceD,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// we expect spacebinding inherited from space-B
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+					// we expect spacebinding inherited from space-C
+					*spacebinding.NewSpaceBinding(spaceCMur, spaceC, spaceC.Name, spacebinding.WithRole("maintainer")),
+					// and the one from space-D
+					*spacebinding.NewSpaceBinding(spaceDMur, spaceD, spaceD.Name, spacebinding.WithRole("admin")),
+				},
+			},
+			"for Space-E": {
+				space:                 spaceE,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{},
+			},
+			"for Space-F": {
+				space:                 spaceF,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// specific one only for this space
+					*spacebinding.NewSpaceBinding(spaceFMur, spaceF, spaceF.Name, spacebinding.WithRole("viewer")),
+				},
+			},
+		}
+		for k, tc := range tests {
+			t.Run(k, func(t *testing.T) {
+
+				// when
+				spaceBindingLister := spacebinding.NewLister(tc.listSpaceBindingsFunc, tc.getSpaceFunc)
+
+				// then
+				spaceBindings, err := spaceBindingLister.ListForSpace(tc.space, []toolchainv1alpha1.SpaceBinding{})
+				if tc.expectedErr != "" {
+					assert.EqualError(t, err, tc.expectedErr)
```

Comment:
I don't see an expected error in any of the test cases. Maybe we can simplify by removeing this if/else statement ?

Reply:
thx, removed it